source/codegen/metadata/nixnetsocket/functions.py

```diff
@@ -163,12 +163,28 @@
             {
                 'direction': 'in',
                 'name': 'node',
-                'type': 'const char[]'
+                'type': 'const char[]',
+                'proto_only': True
+            },
+            {
+                'direction': 'in',
+                'name': 'node_api',
+                'type': 'const char[]',
+                'hardcoded_value': 'request->node() == "" ? nullptr : node',
```

Comment:
I'm not crazy about this but something like this was needed to get the driver-api tests to pass. They would fail when passing an empty string instead of nullptr (the C API tests pass nullptr). This seemed like the easiest way to generate the necessary conditional. I could also look into something like a 'nullptr-if-empty' tag for strings and have it generate this but that might be overkill if it's not widely used.

Reply:
I think the `optional` tag is logically a good match for this, but we'd need new code. That's another case where we treat "empty-as-null", but onyl with linked arrays. I thought I ran into this with other methods but they all accepted `""` to mean the same thing as `nullptr`. I guess that's not the case here?

I think this is OK though.

Reply:
Yeah, so it was complaining in the `NumericPortTranslation_GetAddrInfo_ReturnsExpectedAddrInfo` test I added before adding this with the `node` argument as an empty string. I think this is because some hint flags were set for the test, specifically `GET_ADDR_INFO_FLAGS_NUMERICHOST`. I've also seen in all other cases that it seems to accept an empty string as `nullptr`.

And yeah, I think some version of `optional` would make sense. But I'm going to leave it as-is for now. If more instances like this crop up, though, we should reconsider.